internal/cmd/stack/list.go

```diff
@@ -23,57 +20,40 @@ func listStacks() cli.ActionFunc {
 		case cmd.OutputFormatTable:
 			return listStacksTable(cliCtx)
 		case cmd.OutputFormatJSON:
-			return listStacksJSON(cliCtx.Context)
+			return listStacksJSON(cliCtx)
 		}
 
 		return fmt.Errorf("unknown output format: %v", outputFormat)
 	}
 }
 
-func listStacksJSON(ctx context.Context) error {
-	var query struct {
-		Stacks []stack `graphql:"stacks" json:"stacks,omitempty"`
+func listStacksJSON(ctx *cli.Context) error {
+	stacks, err := searchStacks(ctx.Context, structs.SearchInput{})
+	if err != nil {
+		return err
 	}
 
-	if err := authenticated.Client.Query(ctx, &query, map[string]interface{}{}); err != nil {
-		return errors.Wrap(err, "failed to query list of stacks")
-	}
-	return cmd.OutputJSON(query.Stacks)
+	return cmd.OutputJSON(stacks)
 }
 
 func listStacksTable(ctx *cli.Context) error {
-	var query struct {
-		Stacks []struct {
-			ID            string   `graphql:"id" json:"id,omitempty"`
-			LockedBy      string   `graphql:"lockedBy"`
-			Name          string   `graphql:"name"`
-			State         string   `graphql:"state"`
-			Labels        []string `graphql:"labels"`
-			TrackedCommit struct {
-				AuthorName string `graphql:"authorName"`
-				Hash       string `graphql:"hash"`
-			} `graphql:"trackedCommit"`
-			WorkerPool struct {
-				Name string `graphql:"name"`
-			} `graphql:"workerPool"`
-		} `graphql:"stacks"`
-	}
-
-	if err := authenticated.Client.Query(ctx.Context, &query, map[string]interface{}{}); err != nil {
-		return errors.Wrap(err, "failed to query list of stacks")
-	}
-
-	sort.SliceStable(query.Stacks, func(i, j int) bool {
-		return strings.Compare(strings.ToLower(query.Stacks[i].Name), strings.ToLower(query.Stacks[j].Name)) < 0
+	stacks, err := searchStacks(ctx.Context, structs.SearchInput{
+		OrderBy: &structs.QueryOrder{
```

Comment:
I wonder if we should follow the same logic as the `default view` in the UI for stacks? I think that one is
```
"orderBy": [
    {
        "field": "starred",
        "direction": "DESC"
    }
]
``

Reply:
Yes, precisely. 👍🏻

Reply:
2d7797ce4b02f3000d234c94d1aa2e0f38c03eb2